cmd/lingo/main.go

```diff
@@ -72,64 +77,96 @@ func run() error {
 			BindAddress: metricsAddr,
 		},
 		HealthProbeBindAddress: probeAddr,
-		LeaderElection:         enableLeaderElection,
-		LeaderElectionID:       "af3bat4f.substratus.ai",
+		// LeaderElection is done in the Autoscaler.
+		LeaderElection: false,
 	})
 	if err != nil {
 		return fmt.Errorf("starting manager: %w", err)
 	}
 
-	fifo := queuemanager.NewFIFOQueueManager(concurrencyPerReplica)
+	clientset, err := kubernetes.NewForConfig(mgr.GetConfig())
+	if err != nil {
+		return fmt.Errorf("clientset: %w", err)
+	}
+
+	const POD_NAME = "POD_NAME"
+	podName := os.Getenv(POD_NAME)
```

Comment:
could we get this value automatically? or have a default value if POD_NAME is not set?

Reply:
I think defaulting to hostname would be good enough, since that's essentially what you're doing in the YAML:
```
        - name: POD_NAME
          valueFrom:
            fieldRef:
              fieldPath: metadata.name
```

Reply:
Good call, changed